### PR TITLE
feat: commandStack.changed event provides context

### LINKED ISSUE #479: Command stack change event should be more detailed.
__Is your feature request related to a problem? Please describe.__

Using the modeler in an environment with it's own implementation of editor features can be quite frustrating. There is no convenient way to tell what the commandStack did, only that it did something unknown. An example of such an environment is VS Code or atom. Such an environment needs context to function correctly.

__Describe the solution you'd like__

~~The commandStack should emit more detailed events, like `commandStack.pushed`, `commandStack.undid` & `commandStack.redid`. Or it could just be added as event data to the `commandStack.changed` event. Additionally, the executed/reverted commands in that logical unit and their context should be included.~~

Every `commandStack.changed` event has an additional property `trigger` that is either `undo`, `redo`, `execute` or `clear`. This allows integrators to distinguish the different change events, i.e. to manage dirty state externally.

__Describe alternatives you've considered__

Overwriting the commandStack module is also an approach I could look into, but plugins shouldn't change core functionality due to compatibility issues. However other than that it would be a very reasonable approach.

I could use the `commandStack.execute` event and a timer to aggregate all commands with the same id, but it would be great to be able to undo immediately, not after timer.

I could just check the internal data of the commandStack module to determine what exactly changed, but this is unreasonable considering I would be practically implementing my own commandstack in the process, just way more hacky.

### PR DESCRIPTION
**Pull request by PabloDons**
__Which issue does this PR address?__

Closes #479 

__Acceptance Criteria__

* [ ] Corresponds to the concept 
* [ ] Corresponds to the design 

__Definition of Done__

* [x] corresponds to [the design principles](https://github.com/bpmn-io/design-principles)
* [x] corresponds to [the code standards](https://github.com/bpmn-io/diagram-js/blob/master/.github/CONTRIBUTING.md#creating-a-pull-request)
* [x] passes Continuous Integration checks
* [x] available as feature branch on GitHub
  * [x] contains the cleaned up commit history
  * [x] commit messages satisfy our [commit message guidelines](https://www.conventionalcommits.org/)
  * [x] a single commit closes the issue via Closes #issuenr

### CODE REVIEW

**Comment by nikku** [comment]
Great. I did add some minor simplifications on top of your PR via https://github.com/bpmn-io/diagram-js/pull/529/commits/43512f4eeac30dfb2f221b1d3ce88a3b39a4ddfc and will go ahead and merge that one after I got your O.K.

Please double check the simplifications I made.

File: lib/command/CommandStack.js [hunk 1]
```diff
@@ -141,6 +142,8 @@ CommandStack.prototype.execute = function(command, context) {
     throw new Error('command required');
   }
 
+  var execution = this._currentExecution;
+  execution.trigger || (execution.trigger = 'execute');
```
Comment (nikku) [inline lib/command/CommandStack.js:146]: Why is this magic needed, if it could be a simple assignment:

```
execution.trigger = 'execute';
```

File: lib/command/CommandStack.js [hunk 2]
```diff
@@ -199,9 +202,10 @@ CommandStack.prototype.canExecute = function(command, context) {
 CommandStack.prototype.clear = function(emit) {
   this._stack.length = 0;
   this._stackIdx = -1;
+  delete this._currentExecution.trigger;
```
Comment (nikku) [inline lib/command/CommandStack.js:205]: Why is this needed?

**Comment by PabloDons** [comment]
I have tested your simplifications and it works perfectly in my use case! 👍

**Comment by nikku** [comment]
Squashed and merged via https://github.com/bpmn-io/diagram-js/pull/529.

Thanks for your contribution.
